client/extensions/woocommerce/app/products/product-header.js

```diff
@@ -13,21 +13,51 @@ import ActionHeader from 'woocommerce/components/action-header';
 import Button from 'components/button';
 import { getLink } from 'woocommerce/lib/nav-utils';
 
-const ProductHeader = ( { onTrash, onSave, isBusy, translate, site, product } ) => {
-	const trashButton = onTrash &&
-		<Button borderless onClick={ onTrash }><Gridicon icon="trash" /></Button>;
+function renderViewButton( product, translate ) {
+	const url = product && product.permalink;
+	return (
+		// TODO: Do more to validate this URL?
+		<a href={ url } className="products__header-view-link" target="_blank" rel="noopener noreferrer">
+			<Button borderless><Gridicon icon="visible" /><span> { translate( 'View' ) } </span></Button>
+		</a>
+	);
+}
 
+function renderTrashButton( onTrash, product, isBusy, translate ) {
+	return onTrash && (
+		<Button borderless scary onClick={ onTrash }>
+			<Gridicon icon="trash" />
+			<span>{ translate( 'Trash' ) } </span>
+		</Button>
+	);
+}
+
+function renderSaveButton( onSave, product, isBusy, translate ) {
 	const saveExists = 'undefined' !== typeof onSave;
 	const saveDisabled = false === onSave;
 
-	const saveButton = saveExists &&
+	const saveLabel = ( product && ! isObject( product.id )
+		? translate( 'Update' )
+		: translate( 'Save & Publish' )
```

Comment:
Hi! I've found a possible matching string that has already been translated **67** times:
`translate( 'Save &amp; Publish' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).